process_basic.go

```diff
@@ -161,6 +161,9 @@ func (p *basicProcess) Wait(ctx context.Context) (int, error) {
 	case <-p.waitProcessed:
 	}
 
+	if !p.info.Successful {
```

Comment:
This is something that has to be done because of tests in `integration_test.go` assuming that unsuccessful executions will trigger this error with this message in calls to `Process.Wait()`.

Honestly, I think this is bad because it is an undocumented and very particular behavior that `Wait()` needs to conform to which doesn't add much value (it should be trivial to call `Process.Info()` after calling `Process.Wait()` and checking `ProcessInfo.Successful`). This little detail of how `Process.Wait()` should be implemented is easy to miss.